src/include/settings/settings_defs.h

```diff
@@ -182,71 +182,81 @@ SETTING_bool(
 )
 
 SETTING_bool(
-    metrics_thread,
+    use_metrics_thread,
     "Use a thread for the metrics sub-system (default: true).",
     true,
     false,
     terrier::settings::Callbacks::NoOp
 )
 
 SETTING_bool(
-    metrics_logging,
+    logging_metrics_enable,
     "Metrics collection for the Logging component (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsLogging
 )
 
 SETTING_bool(
-    metrics_transaction,
+    transaction_metrics_enable,
     "Metrics collection for the TransactionManager component (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsTransaction
 )
 
 SETTING_bool(
-    metrics_gc,
+    gc_metrics_enable,
     "Metrics collection for the GarbageCollector component (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsGC
 )
 
 SETTING_bool(
-    metrics_query_trace,
+    query_trace_metrics_enable,
     "Metrics collection for Query Traces (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsQueryTrace
 )
 
 SETTING_bool(
-    metrics_execution,
+    execution_metrics_enable,
     "Metrics collection for the Execution component (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsExecution
 )
 
 SETTING_bool(
-    metrics_pipeline,
+    pipeline_metrics_enable,
     "Metrics collection for the ExecutionEngine pipelines (default: false).",
     false,
     true,
     terrier::settings::Callbacks::MetricsPipeline
 )
 
+SETTING_int(
+    pipeline_metrics_interval,
+    "Sampling rate of metrics collection for the ExecutionEngine pipelines (default: 0).",
```

Comment:
What does 0/1/2/... mean? 0 = nothing, 10 = 100%? Could you update the comment here?